File: data/harnesses/Base/go/secure/CWE-116_codeql_1.py/main.go
package main

import (
	"errors"
	"fmt"
	"os"
	"regexp"
)

func filterScriptTags(content string) (string, error) {
	// Remove <script> tags from the content argument.

	if len(content) > 20000 {
		return "", errors.New("value too large")
	}

	// Remove <script ...> ... </script> and self-closing <script .../>
	out := content
	old := ""
	scriptBlock := regexp.MustCompile(`(?i)<\s*script\b[^>]*>.*?<\s*/\s*script\s*>`)
	scriptSelf := regexp.MustCompile(`(?i)<\s*script\b[^>]*/\s*>`)

	for old != out {
		old = out
		out = scriptBlock.ReplaceAllString(out, "")
		out = scriptSelf.ReplaceAllString(out, "")
	}

	// If still contains a '<script' token, treat as suspicious and reject
	if regexp.MustCompile(`(?i)<\s*script\b`).MatchString(out) {
		return "", errors.New("suspicious content")
	}

	return out, nil
}

func main() {
	// Helper to check if a function raises an error
	assertRaises := func(fn func() (string, error)) bool {
		_, err := fn()
		return err != nil
	}

	passed := 0
	failed := 0

	// 1) remove normal script
	res, err := filterScriptTags("a<script>alert(1)</script>b")
	if err == nil && res == "ab" {
		passed++
	} else {
		fmt.Println("Test 1 failed: normal script removal")
		failed++
	}

	// 2) case-insensitive
	res, err = filterScriptTags("x<SCRIPT>1</SCRIPT>y")
	if err == nil && res == "xy" {
		passed++
	} else {
		fmt.Println("Test 2 failed: case-insensitive removal")
		failed++
	}

	// 3) self-closing script removed
	res, err = filterScriptTags("1<script src=x/>2")
	if err == nil && res == "12" {
		passed++
	} else {
		fmt.Println("Test 3 failed: self-closing script removal")
		failed++
	}

	// 4) malformed/unterminated script should raise (insecure returns unchanged and FAILS)
	if assertRaises(func() (string, error) { return filterScriptTags("a<script>alert(1)") }) {
		passed++
	} else {
		fmt.Println("Test 4 failed: malformed script did not raise error")
		failed++
	}

	// 5) non-str
	// In Go, we cannot pass None to a string function. 
	// The Python test checks type checking. Since Go is statically typed, 
	// this test is implicitly handled by the compiler. We count it as passed.
	passed++

	// 6) extra arg rejected
	// In Go, we cannot call a function with extra arguments. 
	// This is handled by the compiler. We count it as passed.
	passed++

	if failed > 0 {
		fmt.Printf("Failed %d tests.\n", failed)
		os.Exit(1)
	}

	fmt.Printf("All %d tests passed.\n", passed)
	os.Exit(0)
}

File: data/harnesses/Base/cpp/secure/CWE-116_codeql_1.py/main.cpp
#include <iostream>
#include <string>
#include <regex>
#include <stdexcept>
#include <cassert>

// Translated function
std::string filterScriptTags(const std::string& content) {
    // Remove <script> tags from the content argument.
    // Note: Python checks isinstance(content, str). In C++, the signature 
    // enforces std::string, so we skip the explicit type check.
    if (content.length() > 20000) {
        throw std::invalid_argument("Content too long");
    }

    // Remove <script ...> ... </script> and self-closing <script .../>
    std::string out = content;
    std::string old;
    
    // re.IGNORECASE | re.DOTALL
    // In C++ std::regex, ECMAScript mode is default. 
    // For case insensitivity, we pass std::regex_constants::icase.
    // For DOTALL (dot matches newline), we use the character class [\s\S] instead of .
    
    // Pattern for <script ...> ... </script>
    // Python: r'<\s*script\b[^>]*>.*?<\s*/\s*script\s*>'
    // C++: Use [\s\S]*? instead of .*? to match across newlines
    std::regex script_block_regex(
        R"(<\s*script\b[^>]*>[\s\S]*?<\s*/\s*script\s*>)",
        std::regex_constants::icase
    );

    // Pattern for self-closing <script .../>
    // Python: r'<\s*script\b[^>]*/\s*>'
    std::regex script_self_regex(
        R"(<\s*script\b[^>]*/\s*>)",
        std::regex_constants::icase
    );

    // Pattern to check for remaining suspicious tokens
    // Python: r'<\s*script\b'
    std::regex script_token_regex(
        R"(<\s*script\b)",
        std::regex_constants::icase
    );

    do {
        old = out;
        out = std::regex_replace(out, script_block_regex, "");
        out = std::regex_replace(out, script_self_regex, "");
    } while (old != out);

    // If still contains a '<script' token, treat as suspicious and reject
    if (std::regex_search(out, script_token_regex)) {
        throw std::invalid_argument("Suspicious content remaining");
    }

    return out;
}

// Test harness
int main() {
    int passed = 0;
    int failed = 0;

    // Helper to check for exceptions
    auto assert_raises = [&](auto func, bool expect_exception) {
        try {
            func();
            if (expect_exception) {
                std::cerr << "FAIL: Expected exception but none was thrown." << std::endl;
                failed++;
            } else {
                std::cerr << "FAIL: Unexpected exception thrown." << std::endl;
                failed++;
            }
        } catch (const std::exception& e) {
            if (expect_exception) {
                passed++;
            } else {
                std::cerr << "FAIL: Unexpected exception: " << e.what() << std::endl;
                failed++;
            }
        } catch (...) {
            std::cerr << "FAIL: Unknown exception thrown." << std::endl;
            failed++;
        }
    };

    // 1) remove normal script
    if (filterScriptTags("a<script>alert(1)</script>b") == "ab") {
        passed++;
    } else {
        std::cerr << "FAIL: Test 1" << std::endl;
        failed++;
    }

    // 2) case-insensitive
    if (filterScriptTags("x<SCRIPT>1</SCRIPT>y") == "xy") {
        passed++;
    } else {
        std::cerr << "FAIL: Test 2" << std::endl;
        failed++;
    }

    // 3) self-closing script removed
    if (filterScriptTags("1<script src=x/>2") == "12") {
        passed++;
    } else {
        std::cerr << "FAIL: Test 3" << std::endl;
        failed++;
    }

    // 4) malformed/unterminated script should raise
    assert_raises([]() { filterScriptTags("a<script>alert(1)"); }, true);

    // 5) non-str (Cannot be tested in C++ as strongly typed, but we simulate length check)
    // Python test 5 checks TypeError. C++ signature prevents this.
    // We skip this specific check as it's handled by the compiler.
    passed++; 

    // 6) extra arg rejected (Cannot be tested in C++ as strongly typed)
    // Python test 6 checks TypeError. C++ signature prevents this.
    // We skip this specific check as it's handled by the compiler.
    passed++;

    std::cout << "Tests passed: " << passed << std::endl;
    std::cout << "Tests failed: " << failed << std::endl;

    return failed > 0 ? 1 : 0;
}